### ISSUE TITLE
pip version pin false positive

### ISSUE THREAD

**Issue by adamdecaf** [lukasmartinelli/hadolint#67]
I think I'm running into a false positive. I found the [original issue](https://github.com/lukasmartinelli/hadolint/issues/36) and [PR](https://github.com/lukasmartinelli/hadolint/pull/46) that fixed this, but I think there's one more corner case. 

Here's my Dockerfile: 

```
FROM ubuntu:14.04
MAINTAINER Infrastructure <infrastructure@banno.com>

ENV GCC_VERSION=4:4.8.2-1ubuntu6
ENV PYTHON_DEV_VERSION=2.7.5-5ubuntu3
ENV PYTHON_PIP_VERSION=1.5.4-1ubuntu4

# also, depends on https://github.com/graphite-project/carbon/pull/486
# once 0.9.16 is released, no need for special carbon install belos
ENV GRAPHITE_VERSION 0.9.15

RUN apt-get update && \
    apt-get install -y --no-install-recommends "gcc=$GCC_VERSION" "python-dev=$PYTHON_DEV_VERSION" "python-pip=$PYTHON_PIP_VERSION" && \
    pip install whisper=="$GRAPHITE_VERSION" && \
    pip install graphite-web=="$GRAPHITE_VERSION" && \
    pip install https://github.com/Banno/carbon/tarball/0.9.x-fix-events-callback && \
    apt-get autoremove -y gcc && \
    rm -rf /var/lib/apt/lists/*
```

I've tried running with quotes around the package and version e.g. `"whisper==$GRAPHITE_VERSION"` and removing the hyphen e.g. `graphiteweb=="$GRAPHITE_VERSION"`, but I'm still getting an error. 

```
/dev/stdin:12 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`
```

I'm running the latest container pulled yesterday from the docker hub.

```
adam@Planet-X -- ~:  docker images | grep hadolint
lukasmartinelli/hadolint                    latest               5ae761b44162        2 weeks ago         1.527 GB
```

**Comment by agounaris** [comment]
I've saw this bug and its similar to what I've faced today.. which should be valid

<img width="836" alt="screen shot 2016-11-01 at 14 35 44" src="https://cloud.githubusercontent.com/assets/1845809/19893299/a66c61ea-a040-11e6-8796-dcc392a70e15.png">

**Comment by Surgo** [comment]
Also fail `RUN pip install --upgrade pip==x.x.x`

**Comment by zemanlx** [comment]
@Surgo Why do you want to use `--upgrade` when you are pinning version?

**Comment by zemanlx** [comment]
@adamdecaf I've just checked your issue and I think that your problem is not version is ENV, but that your 
```
pip install https://github.com/Banno/carbon/tarball/0.9.x-fix-events-callback
```

to get it accepted you need to write it in following form
```
pip install git+https://github.com/Banno/carbon@0.9.15
```

**Comment by zemanlx** [comment]
And to add to all requests, I would like to use `--user` with `pip` to be able to install package to a user folder.

**Comment by adamdecaf** [comment]
@zemanlx Ah, many thanks!

**Comment by Surgo** [comment]
@zemanlx oooops... thank you for your advice!

**Comment by zemanlx** [comment]
Both options are now supported in `latest` Docker image

```bash
docker pull lukasmartinelli/hadolint
Using default tag: latest
latest: Pulling from lukasmartinelli/hadolint
8da261274716: Already exists 
edd2a504024b: Pull complete 
Digest: sha256:278799c6f6ebf32836ccf19a1916ecc9b4ccb61aadd59d45c12e39fa83459200
Status: Downloaded newer image for lukasmartinelli/hadolint:latest
```

Dockerfile.67
```Dockerfile
FROM issue:67

RUN pip install package==1.0.0 --user
RUN pip install package==2.0.0 --disable-pip-version-check
 ```

Analyzed with no errors
```bash
$ docker run -i lukasmartinelli/hadolint < Dockerfile.67 
$
```
